docker-compose.swarm-stack.yml

```diff
@@ -0,0 +1,215 @@
+version: '3'
+services:
+  redis:
+    restart: always
+    image: sameersbn/redis:4.0.9-1
+    command:
+    - --loglevel warning
+    volumes:
+    - gitlab-redis:/var/lib/redis:Z
+
+  postgresql:
+    restart: always
+    image: sameersbn/postgresql:10
+    volumes:
+    - gitlab-postgres:/var/lib/postgresql:Z
+    environment:
+    - DB_USER=gitlab
+    - DB_PASS=password
+    - DB_NAME=gitlabhq_production
+    - DB_EXTENSION=pg_trgm
+
+  registry:
+    image: registry:2
+    restart: always
+    depends_on:
+      - gitlab
+    volumes:
+      - gitlab-registry:/registry
+      - gitlab-certs:/certs
+    environment:
+      - REGISTRY_LOG_LEVEL=info
+      - REGISTRY_STORAGE_FILESYSTEM_ROOTDIRECTORY=/registry
+      - REGISTRY_AUTH_TOKEN_REALM=https://git.example.com/jwt/auth
+      - REGISTRY_AUTH_TOKEN_SERVICE=container_registry
+      - REGISTRY_AUTH_TOKEN_ISSUER=gitlab-issuer
+      - REGISTRY_AUTH_TOKEN_ROOTCERTBUNDLE=/certs/registry.crt
+      - REGISTRY_STORAGE_DELETE_ENABLED=true
+    deploy:
+      labels:
+        - "traefik.frontend.rule=Host:registry.example.com"
+        - "traefik.enable=true"
+        - "traefik.port=5000"
+        - "traefik.tags=traefik-public"
+        - "traefik.docker.network=traefik-public"
+        # Traefik service that listens to HTTP
+        - "traefik.redirectorservice.frontend.entryPoints=http"
+        - "traefik.redirectorservice.frontend.redirect.entryPoint=https"
+        # Traefik service that listens to HTTPS
+        - "traefik.webservice.frontend.entryPoints=https"
+    networks:
+      - traefik-public
+      - default
+  gitlab:
+    restart: always
+    image: sameersbn/gitlab:11.6.0
+    depends_on:
+    - redis
+    - postgresql
+    ports:
+    - "22:22"
+    volumes:
+    - gitlab-gitlab:/home/git/data:Z
+    - gitlab-certs:/certs
+    deploy:
+      labels:
+        - "traefik.frontend.rule=Host:git.example.com"
```

Comment:
For an easier management it would be more nice that use an variable to get the consistency other multiple values.

This means it would look like this
```
 - "traefik.frontend.rule=Host:${GITLAB_HOST:-git.example.com}"
```

This can be now overriden in two different ways. 

1. Pass it directly with command `GITLAB_HOST=2-git.example.com` docker stack deploy -c docker-compose.swarm-stack.yml
2. You can provide an file called `.env` and write the environment variables directly
```bash
#.env
GITLAB_HOST=2-git.example.com
```

Reply:
Awesome.

I've been using that recently but imagined you wouldn't like having extra files for this.